src/storage/heap_file.c

```diff
@@ -16725,6 +16725,19 @@ heap_set_autoincrement_value (THREAD_ENTRY * thread_p, HEAP_CACHE_ATTRINFO * att
 	  if (prm_get_integer_value (PRM_ID_SUPPLEMENTAL_LOG) > 0)
 	    {
 	      OID serial_obj_oid = att->auto_increment.serial_obj.load ().oid;
+
+	      LOG_TDES *tdes = LOG_FIND_CURRENT_TDES (thread_p);
+	      if (tdes)
+		{
+		  if (!tdes->has_supplemental_log)
```

Comment:
I'm not sure that `(tdes != NULL)` is always true. because the caller for this function (heap_set_autoincrement_value()) check whether tdes is null or not. 

When a thread is allocated and initialized to perform a transaction, log_tdes and tran_index are assigned, so there is no case where the tran_index corresponding to the thread cannot be found (tdes==null). It seems nice to add an assert.